server/src/main/java/org/elasticsearch/index/translog/BufferedChecksumStreamInput.java

```diff
@@ -101,7 +102,11 @@ public void reset() throws IOException {
 
     @Override
     public int read() throws IOException {
-        return readByte() & 0xFF;
+        try {
+            return readByte() & 0xFF;
+        } catch (EOFException e) {
```

Comment:
Can you confirm that current clients of this read() api do not rely on OEFException ?

Reply:
Indeed I see that according to [InputStream documentation](https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#read--) it should return -1 for EOF. I also searched a bit for uses of this function in the code, and so far, I couldn't quickly say that someone depended on the behavior of a EOFException being thrown from this function. However, I agree with @volodk85 that we should have a confirmation from @Tim-Brooks on this before merging.

Reply:
This stream type is used to write translog operations which are sized hence why the improper stream behavior has not mattered up until this point. We read a fixed size opposed to consuming stream until -1. This incorrect exception behavior introduces problems if we want to expand the usage of this stream type (`Streams.copy(input, output, false);` is where I identified this) as throwing an exception is not the correct way to signal an end of stream.